Formula/qscintilla2.rb

```diff
@@ -11,26 +12,38 @@ class Qscintilla2 < Formula
     sha256 "a2a4e459bc424996597fdd75bd597c279e157f8818c6e106107dce516ab46c24" => :mavericks
   end
 
-  option "without-plugin", "Skip building the Qt Designer plugin"
+  option "with-qt5", "Build with Qt5 support"
+  option "with-plugin", "Build the Qt Designer plugin"
   option "without-python", "Skip building the Python bindings"
 
   depends_on :python => :recommended
   depends_on :python3 => :optional
 
-  if build.with? "python3"
-    depends_on "pyqt" => "with-python3"
-  elsif build.with? "python"
-    depends_on "pyqt"
+  if build.with?("python") && build.with?("python3")
+    depends_on "sip" => "with-python3" # python2 is recommended
+    depends_on "pyqt5" => "with-python" if build.with?("qt5") # python3 is recommended
+    depends_on "pyqt" => "with-python3" if build.without?("qt5") # python2 is recommended
+  elsif build.with?("python")
+    depends_on "sip"
+    depends_on "pyqt5" => "with-python" if build.with?("qt5")
+    depends_on "pyqt" if build.without?("qt5")
+  elsif build.with?("python3")
+    depends_on "sip" => "with-python3"
+    depends_on "pyqt5" if build.with?("qt5")
+    depends_on "pyqt" => "with-python3" if build.without?("qt5")
   else
-    depends_on "qt"
+    depends_on "qt5" => :optional
+    depends_on "qt" => :recommended
   end
 
   def install
-    # On Mavericks we want to target libc++, this requires an
-    # unsupported/macx-clang-libc++ flag.
     if ENV.compiler == :clang && MacOS.version >= :mavericks
```

Comment:
Could do `spec = if ENV.compiler == :clang && MacOS.version >= :mavericks` and drop the `spec =` below.